pkg/queue/priority_queue.go

```diff
@@ -0,0 +1,156 @@
+package queue
```

Comment:
Need license.

Reply:
BTW, we may move it in the `/pkg/cache`.

Reply:
it is ok for package